### ISSUE TITLE
i18n RTL: Angular sets <html lang="..."> but not dir="ltr/rtl"

### ISSUE THREAD

**Issue by anisabboud** [angular/angular#44235]
### Which @angular/* package(s) are relevant/releated to the feature request?

localize

### Description

When using Angular i18n (by setting `"i18n" -> "locales"` in `angular.json`), Angular automatically updates the rendered `<html>` element to include the `lang=".."` attribute based on the currently served/built locale, which is great!

However, when the language is RTL (such as Arabic/Hebrew/Persian), Angular doesn't add `dir="rtl"` attribute on the <html> element.
Can this be done, automatically or through some flag (similarly to [`baseHref`](https://github.com/angular/angular-cli/blob/master/packages/angular/cli/lib/config/workspace-schema.json#L298))?

### Proposed solution

Adding `rtl` flag to the `angular.json` schema, which sets `<html dir="rtl">`:
```json
      "i18n": {
        "sourceLocale": { "code": "en", "baseHref": "" },
        "locales": {
          "ar": { "translation": "src/i18n/messages.ar.xlf", "rtl": true },
          "he": { "translation": "src/i18n/messages.he.xlf", "rtl": true }
        }
      },
```

### Alternatives considered

Workarounds considered:
1. `@Inject(LOCALE_ID) public locale: string` in `app.component`, then set `[dir]="locale === 'ar' ? 'rtl' : 'ltr'"` on the root.
    **Problem:** Doesn't cover Material dialogs, since they're under document `body`, not under `app.component` subtree.

2. `@Inject(DOCUMENT) private document: Document` and set: `document.documentElement.dir = getLocaleDirection(locale);`
    **Problem:** Even though that's supposed to cover dialogs, dialogs are rendered messed up. I guess Angular Material's dialog container reads the `html dir` before it gets updated by app.component or something (because if you set `dir="rtl"` manually in `index.html` instead of dynamically, dialogs work just fine).

3. `<html i18n-dir dir="rtl">` in `index.html`.
    **Problem:** Doesn't work because index.html isn't a component, so isn't covered by Angular i18n.

4. CSS override `html[lang=ar] .cdk-global-overlay-wrapper { direction: rtl; }`.
    **Problem:** [`flex-layout`](https://github.com/angular/flex-layout) margins inside the dialog get flipped wrong.

5. Having different `index.html` files for different languages.
    **Problem:** Major code duplication just to change one attribute.

6. Adding a `<head>` script to set the `html dir` attribute based on the `lang` attribute:
  `<script>document.documentElement.dir = ['ar', 'he'].includes(document.documentElement.lang) ? 'rtl' : 'ltr';</script>`
  This seems to work best so far (didn't notice issues), but relies on run-time Javascript instead of setting this AOT in build-time.

To sum up,
- I'm looking for a way to inject `dir="rtl"` into `index.html` in build time, based on the locale defined in `angular.json`.
- Is there a good solution that I'm not aware of to this common RTL issue? 
- If not, perhaps adding an `rtl` flag to `angular.json` locales?

**Comment by alan-agius4** [comment]
Duplicate of https://github.com/angular/angular-cli/issues/16047
